File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.rightView(root, result, 0)
        return result

    def rightView(self, root: Optional[TreeNode], result: List[int], level: int):
        if not root:
            return
        
        if len(result) == level:
            result.append(root.val)
        
        self.rightView(root.right, result, level + 1)
        self.rightView(root.left, result, level + 1)

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> rightview = new ArrayList<>();
        rightView(root, rightview, 0);
        return rightview;
    }

    public void rightView(TreeNode root, List<Integer> result, int level) {
        if (root == null)
            return;
        
        if (result.size() == level)
            result.add(root.val);
        
        rightView(root.right, result, level + 1);
        rightView(root.left, result, level + 1);
    }
}   